lib/datadog/grok/src/matchers/date.rs

```diff
@@ -24,8 +24,16 @@ pub fn convert_time_format(format: &str) -> std::result::Result<String, String>
                 // second of minute
                 's' => time_format.push_str("%S"),
                 // fraction of second
-                'S' => time_format.push_str("%3f"),
+                'S' => {
+                    if time_format.ends_with('.') {
+                        time_format.remove(time_format.len() - 1); // drop .
```

Comment:
You can use [pop](https://doc.rust-lang.org/std/string/struct.String.html#method.pop) to drop the last character. I think that would be clearer.

Reply:
definitely, thanks!